design/src/rfcs/rfc0027_new_home_for_cred_cache_and_builder_api_consistency.md

```diff
@@ -0,0 +1,182 @@
+
+Note that the same set of problems exists for a service client's `config::Builder`. Such a `Builder` has the methods `credentials_provider` and `set_credentials_provider`. Therefore, it is reasonable to expect that the `Builder` will also have the methods `credentials_cache` and `set_credentials_cache`, except that the assumption does not hold as the service client crate, being outside the `aws-config` crate, cannot create a credentials cache, unable to pass it to the said methods.
+
+To summarize, these are the problems encountered as a result of implementation spike for the selected design in the preceding RFC:
+- `ConfigLoader` has `credentials_provider` and `credentials_cache`, but that pair of methods does not seem to behave consistently in `aws_types::sdk_config::Builder` and every service client's `config::Builder`.
+- A credentials cache can only be created within the `aws-config` crate.
+
+Proposed Solutions
+------------------
+In this section we will talk about our proposed solution separately for each of the bullet points from the previous section, one for API consistency and one for creating credentials caches.
+
+### API consistency for `credentials_provider` and `credentials_cache` in other builders
+We propose to have `sdk_config::Builder` and service clients' `config::Builder`s match `ConfigLoader` and take the `credentials_cache` method to make it consistent all the way through.
+
+- :+1: The user can expect a pair of `credentials_provider` and `credentials_cache` methods behaves consistently across `ConfigLoader`,  `sdk_config::Builder` and service clients' `config::Builder`s.
+- :-1: It will require some duplicate code across those places, although it's rather an insignificant amount.
+
+### Creating credentials caches outside the `aws-config` crate
+We propose to move types and traits for credentials caching to a separate crate (tentatively called `aws-XXX`). Specifically, the following items will be moved to `aws-XXX`:
+- All items in `aws_types::credentials` and their dependencies
+- All items in `aws_config::meta::credentials` and their dependencies
+
+The new crate will be the home for `ProvideCachedCredentials` discussed above as well as `CredentialsCache` with its `Lazy` variant and builder. Crucially, `CredentialsCache::create_cahe` will be marked as `pub`. The `aws-XXX` crate itself depends on `aws_smithy_types` and `aws_smithy_async`, among others, for the items in it to be able to compile. There is one wrinkle, however, that is `LazyCredentialsCache` depends on `aws_types::os_shim_internal::TimeSource`; it does not feel right to move `TimeSource` to `aws-XXX` just because `LazyCredentialsCache` requires it. There are two ways to work around it.
```

Comment:
```suggestion
The new crate will be the home for `ProvideCachedCredentials` discussed above as well as `CredentialsCache` with its `Lazy` variant and builder. Crucially, `CredentialsCache::create_cache` will be marked as `pub`. The `aws-XXX` crate itself depends on `aws_smithy_types` and `aws_smithy_async`, among others, for the items in it to be able to compile. There is one wrinkle, however, that is `LazyCredentialsCache` depends on `aws_types::os_shim_internal::TimeSource`; it does not feel right to move `TimeSource` to `aws-XXX` just because `LazyCredentialsCache` requires it. There are two ways to work around it.
```